src/waveform/waveformwidgetfactory.h

```diff
@@ -64,8 +64,9 @@ class WaveformWidgetFactory : public QObject, public Singleton<WaveformWidgetFac
 
     bool setConfig(UserSettingsPointer config);
 
-    //creates the waveform widget and bind it to the viewer
-    //clean-up every thing if needed
+    // creates the waveform widget using the type set with setWidgetType
+    // and binds it to the viewer.
+    // deletes older widget and resets positions to config defaults
```

Comment:
I propose that we use 3 slashes for comments that document whole functions, so that they are picked up by doxygen.  Please change this for the other occurrences, too.

```suggestion
    /// Creates the waveform widget using the type set with setWidgetType
    /// and binds it to the viewer.
    /// Deletes older widget and resets positions to config defaults
```